modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/utils/client_extensions/loadClientExtensions.ts

```diff
@@ -42,6 +42,8 @@ export default function loadClientExtensions(
 			}
 		);
 
-		Promise.all(promises).then(onLoad);
+		Promise.all(promises)
+			.then(onLoad)
+			.catch(() => onLoad([]));
```

Comment:
We're gracefully failing here, showing no CETs if fetching fails. I think this is a better behavior, without the change editor will be stuck forever loading. This scenario is actually quite likely.

Reply:
Mmmm, not sure if I get it, but I think that if you catch and return an empty array there won't be any way for callers of this method to detect if anything went wrong.

So I'm OK with not showing spinners but I think that the `loadClientExtensions` method, being so low level and so generic, should refrain from doing this stuff and simply throw the exception to the upper level.

Maybe we can add some documentation to the method saying that it may throw if something fails so that people don't forget to catch the exception :thinking: .

Reply:
Maybe..., I'm changing my mind here...

Given that JS is so lenient with everything and doesn't require users to catch anything (like civilized languages like Java, Rust, etc. do) maybe it's better not to throw and if in the future we need to distinguish between errors and real empty returns we can change the function or add a new one.

Reply:
In fact, wouldn't we want to support the case where the caller requests multiple CXs and we are able to load only some of them in the method?

Maybe we could return an array of objects containing the loaded CXs or an error... :thinking: 

Something along the lines of: `ClientExtensionReturn: { impl?: function ; error?: Error }`

What do you think?

Reply:
> In fact, wouldn't we want to support the case where the caller requests multiple CXs and we are able to load only some of them in the method?

That sounds like the best solution to me! I'll make the change.

Just a small note about 
> if you catch and return an empty array there won't be any way for callers of this method to detect if anything went wrong

The current solution also has this problem. On failure, `onLoad` is not invoked. Returning an empty array would be a small improvement, but I agree, not ideal. I far more like your suggestion for full success report.

Reply:
>The current solution also has this problem. On failure, onLoad is not invoked.

True. I was assuming `loadClientExtensions` was a Promise based API, but it's a callback one :+1:

Reply:
@izaera it looks like [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) is exactly what we want, but it requires es2020 compiler for `frontend-js-web`. Is this easily doable? If not, I can probably achieve the same behavior manually.